src/c-writer.cc

```diff
@@ -2536,11 +2536,7 @@ void CWriter::Write(const ExprList& exprs) {
         const ElemSegment* src_segment =
             module_->GetElemSegment(inst->segment_index);
 
-        // TODO: This should be an assert because the validator should catch
-        // this, but currently it doesn't.
-        if (dest_table->elem_type != src_segment->elem_type) {
```

Comment:
Fyi, this converts a release assert into a debug assert. Is this by design?

Reply:
Yes, I think so.  The reason is that the validator should already ensure this.  Checking it again in debug builds maybe makes sense just for a sanity check (and as an aid to the reader), but it should be strictly unnecessary.